File: app/src/main/java/id/ac/ui/cs/mobileprogramming/nabilafv/rememberyourlord/repository/QuranTextRepository.java
package id.ac.ui.cs.mobileprogramming.nabilafv.rememberyourlord.repository;

import android.app.Application;

import androidx.lifecycle.LiveData;

import java.util.List;

import id.ac.ui.cs.mobileprogramming.nabilafv.rememberyourlord.dao.QuranTextDao;
import id.ac.ui.cs.mobileprogramming.nabilafv.rememberyourlord.db.QuranTextDatabase;
import id.ac.ui.cs.mobileprogramming.nabilafv.rememberyourlord.model.QuranText;

public class QuranTextRepository {

    private QuranTextDatabase quranTextDatabase;
    private QuranTextDao quranTextDao;

    public QuranTextRepository(Application application) {
        quranTextDatabase = QuranTextDatabase.getInstance(application);
        quranTextDao = quranTextDatabase.quranTextDao();
    }

    public LiveData<QuranText> getQuranText(int id) {
        return quranTextDao.getQuranText(id);
    }

    public LiveData<List<QuranText>> getAllQuranTexts() {
        return quranTextDao.fetchAllQuranTexts();
    }
}

File: app/src/main/java/id/ac/ui/cs/mobileprogramming/nabilafv/rememberyourlord/repository/QuranTextRepository.kt
package id.ac.ui.cs.mobileprogramming.nabilafv.rememberyourlord.repository

import androidx.lifecycle.LiveData
import id.ac.ui.cs.mobileprogramming.nabilafv.rememberyourlord.dao.QuranTextDao
import id.ac.ui.cs.mobileprogramming.nabilafv.rememberyourlord.model.QuranText
import javax.inject.Inject

class QuranTextRepository @Inject constructor(private val quranTextDao: QuranTextDao) :
    QuranTextRepositoryInterface {
    override fun getQuranText(id: Int): QuranText {
        return quranTextDao.getQuranText(id)
    }

    override fun getAllQuranTexts(): List<QuranText> {
        return quranTextDao.fetchAllQuranTexts()
    }
}